File: Day348_2025/Q348_Leetcode.py
class Solution:
    def numberOfWays(self, corridor: str) -> int:
        MOD = 10**9 + 7
        count = 1
        seats = 0
        previousPairLast = -1

        for i, c in enumerate(corridor):
            if c == 'S':
                seats += 1

                if seats == 2:
                    previousPairLast = i
                    seats = 0
                elif seats == 1 and previousPairLast != -1:
                    count = (count * (i - previousPairLast)) % MOD

        if seats == 1 or previousPairLast == -1:
            return 0

        return count

File: Day348_2025/Q348_Leetcode.cpp
class Solution {
public:
    int numberOfWays(string corridor) {
        // Q. Number of Ways to Divide a Long Corridor
        const int MOD = 1e9 + 7;
        long count = 1;
        int seats = 0;
        int previousPairLast = -1;
        for (int index = 0; index < corridor.length(); index++) {
            if (corridor[index] == 'S') {
                seats += 1;
                if (seats == 2) {
                    previousPairLast = index;
                    seats = 0;
                }

                else if (seats == 1 && previousPairLast != -1) {
                    count *= (index - previousPairLast);
                    count %= MOD;
                }
            }
        }

        if (seats == 1 || previousPairLast == -1) {
            return 0;
        }

        return (int) count;
    }
};